config_system/config_system/general.py

```diff
@@ -39,30 +39,55 @@ def is_visible(config):
     return expr.condexpr_value(e)
 
 
-def enforce_dependent_values(auto_fix=False):
+def enforce_dependent_values(stage, fix_bools=False,
+                             error_level=logging.WARNING):
     """
     Check that values are consistently set, specifically with respect
     to whether the dependencies are correct.
 
-    For boolean values we only reset to n if auto_fix is
-    True. Otherwise we raise an exception (the Mconfig needs to be
-    fixed).
-
-    For string and int values, if the configuration value is not
-    enabled, reset them to "" and 0 respectively. This is different to
-    the bool case as we need to reset default values.
+    This function is expected to be called when we expect the values
+    should be consistent. i.e. after reading in a new config, or prior
```

Comment:
This reads weird. Maybe something like "This function should be called when we expect the values to be consitent"?